fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/predicates/RecordTypeValue.java

```diff
@@ -107,4 +109,16 @@ public boolean isFunctionallyDependentOn(@Nonnull final Value otherValue) {
         }
         return false;
     }
+
+    @Nonnull
+    @Override
+    public Type getResultType() {
+        return Type.primitiveType(Type.TypeCode.LONG); // eval returns a Tuple-friendly record type key which is Long.
+    }
+
+    @Nonnull
+    @Override
+    public String explain(@Nonnull final Formatter formatter) {
+        return "recordTypeKey(" + alias + ")";
```

Comment:
`recordType()` -- we should largely, unless for special things, stick with the xxx for `xxxValue`